### PR TITLE
[NETBEANS-5327] Added the option to open the selected file or directory in the system file explorer

### LINKED ISSUE #5327: Open folder on system file explorer
### Description

I think it should be good to open project foldes at system file explorer (Like the Windows Explorer on Windows). 
Several time we want to make it, but we need to find manually the folder throug the system explorer, what seems to be annoying.

### Use case/motivation

Do right click on the folder, select Show at system explorer option

### Related issues

_No response_

### Are you willing to submit a pull request?

No

### Code of Conduct

Yes

### PR DESCRIPTION
**Pull request by Tnnienn**
Closes #5327 

Hi! first timer here!

I implemented this solution that opens the system file explorer using Desktop.getDesktop().open(directory); instruction

### CODE REVIEW

File: ide/editor.actions/src/org/netbeans/modules/editor/actions/OpenInFileExplorerAction.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            return;
+        }
+
+        Node node = activatedNodes[0];
+        FileObject fo = node.getLookup().lookup(FileObject.class);
+
+        final String path = (fo.isFolder()) ? fo.getPath() : fo.getParent().getPath();
+
+        SwingUtilities.invokeLater(() -> {
+            try {
```
Comment (lbownik) [inline ide/editor.actions/src/org/netbeans/modules/editor/actions/OpenInFileExplorerAction.java:63]: how about moving try-catch block into openDirectory method?

**Comment by neilcsmith-net** [comment]
Hi @Tnnienn - always good to see a first time contributor!  :smile: 

As mentioned at https://github.com/apache/netbeans/issues/5327#issuecomment-1398323497 we shouldn't need to add a new action for this though, just register the existing one in new places.  Did you try that first?  Did it not work?

**Comment by Tnnienn** [comment]
my bad, didn't read the comment. i will try😃

**Comment by Tnnienn** [comment]
I updated the PR, making the already existing action visible in the right click menu!!

**Comment by mbien** [comment]
hi @Tnnienn
there are some unrelated changes in project.xml and Bundle.properties which are traces from your first attempt. 
The easiest way to get rid of those would be to simply remove the first two commits.

Once done, please make sure the commit has a valid author name, you can check it by looking at https://github.com/apache/netbeans/pull/5770.patch

feel free to force push into this PR. Result should be a single commit.

**Comment by Tnnienn** [comment]
should be done!!

**Comment by mbien** [comment]
just a thought for discussion (don't change it yet), how about this:
```
@ActionReferences ({
    @ActionReference(path = "UI/ToolActions/Files", position = 2045),
    @ActionReference(path = "Projects/Actions", position = 101) 
})
```
this would essentially register it everywhere and put it in the tools sub menu just like the [terminal action](https://github.com/apache/netbeans/blob/6921b8a4cbf0b1ab786db81c07cac6557ad1232b/ide/dlight.terminal/src/org/netbeans/modules/dlight/terminal/action/OpenInTerminalAction.java#L55).
We have more requests like this one here, e.g #5735. So we could have everything at the same spot without cluttering the top level menu.

In my mind, "Open in System", "Open in Terminal" and "Copy Filesystem Path" are somewhat related to each other.

@neilcsmith-net ping

**Comment by Tnnienn** [comment]
Yes, the first time i wrote it like that, idk why i changed it😅😅

**Comment by mbien** [comment]
@eirikbakke @matthiasblaesing @neilcsmith-net we could squeeze this little feature in if we figure out where to put it :)

I am leaning towards enabling it everywhere (folders, files, projects etc) and hiding it in the tools sub menu, see https://github.com/apache/netbeans/pull/5770#issuecomment-1496485558

**Comment by eirikbakke** [comment]
Yeah, agree that the Tools submenu is a good place to put this!

**Comment by mbien** [comment]
@Tnnienn would you like to change it to the version I posted there: https://github.com/apache/netbeans/pull/5770#issuecomment-1496485558

**Comment by Tnnienn** [comment]
Sure, I'll change it tomorrow in the afternoon (rn it's 00:07 in Italy)

**Comment by Tnnienn** [comment]
@mbien Done!!

**Comment by mbien** [comment]
@Tnnienn commit has now your nick name, pretty sure it was the full name last time https://patch-diff.githubusercontent.com/raw/apache/netbeans/pull/5770.patch

**Comment by Tnnienn** [comment]
you're right now i'm on my laptop and git wasn't configured. I'll fix it rn

**Comment by Tnnienn** [comment]
should be done!

**Comment by Tnnienn** [comment]
should be done @mbien

**Review by mbien — APPROVED** [review]
thanks

**Comment by mbien** [comment]
@Tnnienn congrats to your first contribution

**Comment by Tnnienn** [comment]
Thanks. I will be happy to contribute more. Step by step maybe into something more consistent.❤️

**Comment by neilcsmith-net** [comment]
Thanks @Tnnienn Turns out the existing action has some issues when included in the global menu, so I've had a go at updating it to work dynamically there.  I think it's still working as before, when it's applicable, but an extra check would be great! :smile:

**Comment by lalo-mx** [comment]
Open in system is also shown for files, is it expected to open the file with Windows/Operating System? It would more useful to open the folder of the file.
image

**Comment by neilcsmith-net** [comment]
It's expected to open the file - from the menu or using `SHIFT-ALT-S`. I find this quite useful!

Directly open file location could also be useful I guess, although you can select the parent in the project / files view and open from there.

**Pull request merged**